index.html

```diff
@@ -4297,19 +4304,60 @@ <h2>JSON Processing</h2>
 implementing the rule above. This can ensure proper term identification,
 typing, and order, when a JSON document is processed as JSON-LD.
           </p>
+          <p>
+However, the intention of the issuer is not preserved, unless the <code>@context</code>
+is applied, and the resulting <a href="#verifiable-credential-graphs">Verifiable Credential Graphs</a>,
+and their exact RDF types, and node and edge structure is understood by the verifier.
+          </p>
 
           <p>
-The rule above guarantees semantic interoperability between JSON and JSON-LD for
-literal JSON keys mapped to URIs by the `@context` mechanism. While JSON-LD
-processors will use the specific mechanism provided and can verify that all
-terms are correctly specified, JSON-based processors implicitly accept the same
-semantics without performing any JSON-LD transformations, but instead by
+These rules alone do not guarantee semantic interoperability between JSON and JSON-LD for
+literal JSON keys mapped to URIs by the `@context` mechanism.
```

Comment:
https://w3c.github.io/vc-data-model/#example-a-simple-example-of-the-contents-of-a-verifiable-presentation

```json
{
  "@context": [
    "https://www.w3.org/ns/credentials/v2",
    "https://www.w3.org/ns/credentials/examples/v2",
    // imagine 50% of the time this is in examples/v2 ^ and 50% of the time its not
    { "name": { "@id": "https://bar.example", "@type": "@id" }} 
  ],
  "type": "VerifiablePresentation",
  "name": "bob",
  "verifiableCredential": [{
    "@context": [
      "https://www.w3.org/ns/credentials/v2",
      "https://www.w3.org/ns/credentials/examples/v2"
    ],
    "id": "http://university.example/credentials/1872",
    "type": ["VerifiableCredential", "ExampleAlumniCredential"],
    "issuer": "https://university.example/issuers/565049",
    "validFrom": "2010-01-01T19:23:24Z",
    "credentialSubject": {
      "id": "did:example:ebfeb1f712ebc6f1c276e12ec21",
      "alumniOf": {
        "id": "did:example:c276e12ec21ebfeb1f712ebc6f1",
        "name": "Example University"
      }
    }
  }]
}
```

```json
{
  "@graph": [
    {
      "@type": "https://www.w3.org/2018/credentials#VerifiablePresentation",
      "https://bar.example": {
        "@id": "bob"
      },
      "https://www.w3.org/2018/credentials#verifiableCredential": {
        "@id": "http://university.example/credentials/1872",
        "@type": [
          "https://www.w3.org/2018/credentials#VerifiableCredential",
          "https://www.w3.org/ns/credentials/examples#ExampleAlumniCredential"
        ],
        "https://www.w3.org/2018/credentials#credentialSubject": {
          "@id": "did:example:ebfeb1f712ebc6f1c276e12ec21",
          "https://www.w3.org/ns/credentials/examples#alumniOf": {
            "@id": "did:example:c276e12ec21ebfeb1f712ebc6f1",
            "https://www.w3.org/ns/credentials/examples#name": "Example University"
          }
        },
        "https://www.w3.org/2018/credentials#issuer": {
          "@id": "https://university.example/issuers/565049"
        },
        "https://www.w3.org/2018/credentials#validFrom": {
          "@type": "http://www.w3.org/2001/XMLSchema#dateTime",
          "@value": "2010-01-01T19:23:24Z"
        }
      }
    },
    {
      "@id": "bob"
    },
    {
      "@id": "did:example:c276e12ec21ebfeb1f712ebc6f1",
      "https://www.w3.org/ns/credentials/examples#name": "Example University"
    },
    {
      "@id": "did:example:ebfeb1f712ebc6f1c276e12ec21",
      "https://www.w3.org/ns/credentials/examples#alumniOf": {
        "@id": "did:example:c276e12ec21ebfeb1f712ebc6f1",
        "https://www.w3.org/ns/credentials/examples#name": "Example University"
      }
    },
    {
      "@id": "http://university.example/credentials/1872",
      "@type": [
        "https://www.w3.org/2018/credentials#VerifiableCredential",
        "https://www.w3.org/ns/credentials/examples#ExampleAlumniCredential"
      ],
      "https://www.w3.org/2018/credentials#credentialSubject": {
        "@id": "did:example:ebfeb1f712ebc6f1c276e12ec21",
        "https://www.w3.org/ns/credentials/examples#alumniOf": {
          "@id": "did:example:c276e12ec21ebfeb1f712ebc6f1",
          "https://www.w3.org/ns/credentials/examples#name": "Example University"
        }
      },
      "https://www.w3.org/2018/credentials#issuer": {
        "@id": "https://university.example/issuers/565049"
      },
      "https://www.w3.org/2018/credentials#validFrom": {
        "@type": "http://www.w3.org/2001/XMLSchema#dateTime",
        "@value": "2010-01-01T19:23:24Z"
      }
    },
    {
      "@id": "https://university.example/issuers/565049"
    }
  ]
}
```

The point is that processing the context leads to an (often but not necessarily always) better interpretation of the issuer or holder's intended claims.

It's not safe to say that you can ignore context processing and "know the same things"... its trust that assuming you preserve context, and context does not change, you can "come to the same conclusions later".

In the same I show, the JSON values are literally mapped differently when the context is applied... We know that context can change, perhaps when eratta are published, this is why we set the cache policy for contexts, and why we care about treating contexts as previously dereferenced.

I the context in the example above changes, the semantic interop changes, including how term definitions are mapped.

The only way the statement holds that committing to a context preserves semantic interop is when the context is immutable, and we know thats not the case, even if its "usually" the case... since this is a security spec, the bar should not be "mostly true", the bar should be "always true", or "not always true".

Reply:
I don't think anyone is saying "ignore context processing"... we're saying "if these N things hold true, then you can guarantee that the semantics are the same". So, all we need to do is list the N things that need to hold true. You seem to be saying that the hash for an "immutable" context can change if there are errata... but even then, software can check hashes to detect that -- perhaps that's the one rule that we'd need to add for the statement around the semantics being the same between static processing vs. dynamic processing to hold?

Reply:
decline to implement feedback, but I think we may be able to adjust text in a future editorial pass